handlebars/src/main/java/com/github/jknack/handlebars/internal/Partial.java

```diff
@@ -120,10 +121,31 @@ protected void merge(final Context context, final Writer writer)
       /** Inline partial? */
       LinkedList<Map<String, Template>> partials = context.data(Context.INLINE_PARTIALS);
       Map<String, Template> inlineTemplates = partials.getLast();
+      Template callee = context.data(Context.CALLEE);
+
+      final boolean pathIsPartialBlock = "@partial-block".equals(path);
+      final Template lastPartialBlock = inlineTemplates.get("@partial-block");
+      final boolean parentIsNotLastPartialBlock = !isCalleeOf(callee, lastPartialBlock);
+
+      if (pathIsPartialBlock && parentIsNotLastPartialBlock) {
+        throw new IllegalArgumentException(
```

Comment:
Already merged, but I'm struggling to figure out why we do fail here, can you explain?

Here is a test case that fails today bc of this check:

```
shouldCompileTo("{{#> dude x=23}}{{#> dude x=12}}{{/dude}}{{/dude}}",
        $("hash", $(), "partials",
            $("dude", "<div {{#if x}}x={{x}}{{/if}}>{{> @partial-block}}</div>")),
        "<div x=23><div x=12></div></div>");
```

Reply:
Hi and thanks for the merge! We'll get back to you about this asap.